app/models/user.server.ts

```diff
@@ -17,7 +24,11 @@ export async function getUserByUsername(username: PrismaUser['username']) {
   }
 
   // Decorate with the user's base domain as well
-  return { ...user, baseDomain: buildUserBaseDomain(username) };
+  return {
+    ...user,
+    baseDomain: buildUserBaseDomain(username),
```

Comment:
This is not going to work either, since the `baseDomain` includes your username, which will need to be the `effectiveUsername` in order to work if I'm impersonating someone.

Need to think about this...

Reply:
So I think we're now good here, and we'll pass in the right username/effectiveUsername when we call this.  If that's not right, please correct me.

Reply:
Maybe I am confused, but we need to change places where `requireUser` is being used? That function calls getUsername, and is used in a few places such as in the action of the new DNS record route:
https://github.com/DevelopingSpace/starchart/blob/14c32e101a22539adbff62ac007011a299cc282e/app/routes/__index/dns-records/new.tsx#L14-L16

Reply:
You're right, I will have to change that as well. I was thinking maybe in a following PR but I'm not sure, I could do it here as well.

Reply:
I would do it here.  It could be smart about looking for either vs needing to be renamed (i.e., `requireUser` and `requireUsername` are still fine).